plumbing/api.go

```diff
@@ -161,6 +161,11 @@ func (api *API) ChainHeadKey() types.TipSetKey {
 	return api.chain.Head()
 }
 
+// ChainSetHead sets `ts` as the new head of this chain iff it exists in the nodes chain store.
```

Comment:
This comment is untrue and needs to be changed or the behavior of `SetHead` needs to be updated to check for membership in the store.

Reply:
I think that functionality should be changed to require the head to exist, i.e. have been validated as part of a valid chain at some point. Changing the parameter here to a `TipSetKey` is consistent with that.